Fix #109, Move variables declared mid-function to the top

thnkslprpt opened the pull request:
**Checklist**
* [x] I reviewed the [Contributing Guide](https://github.com/nasa/osal/blob/main/CONTRIBUTING.md).
* [x] I signed and emailed the appropriate [Contributor License Agreement](https://github.com/nasa/cFS/blob/main/CONTRIBUTING.md#contributor-license-agreement-cla) to GSFC-SoftwareRelease@mail.nasa.gov and copied cfs-program@lists.nasa.gov.

**Describe the contribution**
Fixes #109 
All variables (that I could find) that were declared mid-function were moved to the top of their respective functions.
 
**Testing performed**
Just the CI GitHub actions so far.

**Expected behavior changes**
Aligns CF with the rest of cFS and the relevant coding guidelines.
Makes it easier and quicker to identify all variables in a function.

**Contributor Info**
@thnkslprpt

dzbaker commented:
@thnkslprpt Can you run the Clang formatter?

thnkslprpt commented:
> @thnkslprpt Can you run the Clang formatter?

Yes OK - I think i got it working now.

chillfig commented on fsw/src/cf_cfdp.c:
```diff
@@ -1454,15 +1454,14 @@ int32 CF_CFDP_PlaybackDir(const char *src_filename, const char *dst_filename, CF
-    /* either there's no transaction (first one) or the last one was finished, so check for a new one */
+    os_dirent_t       dirent;
+    CF_Transaction_t *pt;
+    int32 status; /* either there's no transaction (first one) or the last one was finished, so check for a new one */
```
Would you please move the comment `/* either there's no transaction (first one) or the last one was finished, so check for a new one */` down 1 line? This comment seems to pertain more broadly to the logic below line 1459 than `status`'s declaration.

thnkslprpt replied:
No problem whatsoever - thanks for catching that.
Should be updated now.

chillfig reviewed (changes requested):
Just a nit-pick on comment placement. Thank you for the contribution!

chillfig commented on fsw/src/cf_cfdp.c:
```diff
@@ -1454,15 +1454,16 @@ int32 CF_CFDP_PlaybackDir(const char *src_filename, const char *dst_filename, CF
-    os_dirent_t dirent;
+    os_dirent_t       dirent;
+    CF_Transaction_t *pt;
+    int32 status;
```
Thank you for the adjustment on the comment! Would you please adjust these lines according to claang format check?: https://github.com/nasa/CF/actions/runs/3300539445/jobs/5445143166

Claang format check wants changes specifically at lines 1458 and 1459

thnkslprpt replied:
Ran it again Justin - not sure why it didn't auto-run the first time.
Cheers

chillfig reviewed (changes requested):
Thank you! Claang formatting needed

thnkslprpt committed:
Fix #109, Move variables declared mid-function to the top

commit 4e1bb977f89fcdd6137fc568fe0a5bb4473b5b64
Author: Avi Weiss <thnkslprpt@gmail.com>
Date:   Mon Dec 26 16:14:26 2022 +1100

    Typo from merge from main

commit 57ab176bd61055a9b86c3fbafd6e3e47927f292d
Merge: e151abd 8368adc
Author: Avi <thnkslprpt@gmail.com>
Date:   Mon Dec 26 15:16:21 2022 +1100

    Merge branch 'main' into fix-109-variable-declarations

commit e151abda58857f5a46192528591a61eedf6595e8
Author: Avi <thnkslprpt@gmail.com>
Date:   Thu Oct 13 18:34:51 2022 +1000

    Fix #109, Move variables declared mid-function to the top

chillfig reviewed (approved):
Re-reviewed. Some initializations are moved along with the declaration line changes, however, these are done to variables whose assignment is unchanged whether mid-function or top.
